### DISCUSSION TITLE
Error [NextFontError]: Failed to fetch `Inter` from Google Fonts.

### DISCUSSION THREAD

**Discussion by xgqfrms** [vercel/next.js#46012] [Help] [answered]
# Error [NextFontError]: Failed to fetch `Inter` from Google Fonts.

> why not use locally google fonts?

![image](https://user-images.githubusercontent.com/7291672/219448724-0962c582-3b92-43a6-80d3-0d6d79caedf0.png)

```sh
$ yarn dev
yarn run v1.22.19
$ next dev
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
event - compiled client and server successfully in 1856 ms (165 modules)
wait  - compiling / (client and server)...
Error [NextFontError]: Failed to fetch `Inter` from Google Fonts.
    at nextFontError (/Users/xgqfrms-mm/Documents/github/ChatGPT/chat-gpt/node_modules/@next/font/dist/utils.js:55:17)
    at downloadGoogleFonts (/Users/xgqfrms-mm/Documents/github/ChatGPT/chat-gpt/node_modules/@next/font/dist/google/loader.js:87:39)
    at /Users/xgqfrms-mm/Documents/github/ChatGPT/chat-gpt/node_modules/next/dist/build/webpack/loaders/next-font-loader/index.js:52:99
    at async Span.traceAsyncFn (/Users/xgqfrms-mm/Documents/github/ChatGPT/chat-gpt/node_modules/next/dist/trace/trace.js:79:20)
error - Failed to download `Inter` from Google Fonts. Using fallback font instead.
event - compiled client and server successfully in 75.4s (194 modules)
wait  - compiling...
wait  - compiling / (client and server)...
event - compiled client and server successfully in 75.2s (183 modules)


```

**Comment by xgqfrms** [comment, marked as answer]
```diff

import Head from 'next/head'
import Image from 'next/image'
- import { Inter } from '@next/font/google'
import styles from '@/styles/Home.module.css'

- const inter = Inter({ subsets: ['latin'] })

```

**Reply by lukethacoder** [reply]
Not really an answer, but does solve the rendering issues.

If you actually need the fonts, it might be worth bumping your Next package version. I'm rocking `v13.3.1` now and it seems to be working fine.

**Comment by xgqfrms** [comment]
## same problems

https://stackoverflow.com/questions/74889692/next-js-example-takes-a-long-time-to-compile-due-to-inter-google-fonts


## `VPN` proxy bug

https://github.com/vercel/next.js/issues/45034

**Comment by smyja** [comment]
same issue, i'm not using a vpn.

**Comment by javierdev0** [comment]
This work for me https://github.com/vercel/next.js/issues/45080#issuecomment-1680665108

**Reply by mapsgeek** [reply]
same

**Comment by vbandaragoda** [comment]
Update your project as follows.

 `tailwind.config.js`

    /** @type {import('tailwindcss').Config} */
    module.exports = {
    content: [
    "./app/**/*.{js,ts,jsx,tsx,mdx}",
    "./pages/**/*.{js,ts,jsx,tsx,mdx}",
    "./components/**/*.{js,ts,jsx,tsx,mdx}",
    
    // Or if using `src` directory:
    "./src/**/*.{js,ts,jsx,tsx,mdx}",
    ],
    sans: [
          "Inter",
          "ui-sans-serif",
          "system-ui",
          "-apple-system",
          "system-ui",
          "Segoe UI",
          "Roboto",
          "Helvetica Neue",
          "Arial",
          "Noto Sans",
          "sans-serif",
          "Apple Color Emoji",
          "Segoe UI Emoji",
          "Segoe UI Symbol",
          "Noto Color Emoji",
        ],
    sans: [
          "Inter",
          "ui-sans-serif",
          "system-ui",
          "-apple-system",
          "system-ui",
          "Segoe UI",
          "Roboto",
          "Helvetica Neue",
          "Arial",
          "Noto Sans",
          "sans-serif",
          "Apple Color Emoji",
          "Segoe UI Emoji",
          "Segoe UI Symbol",
          "Noto Color Emoji",
        ],
    },
    },
    },
    plugins: [],
    };


 `next.config.js`

    /** @type {import('next').NextConfig} */
     const nextConfig = {
      swcMinify: true,
      };

      module.exports = nextConfig;


`globals.css`

    @import 'tailwindcss/base';
    @import 'tailwindcss/components';
    @import 'tailwindcss/utilities';

     html,
     body {
      padding: 0;
      margin: 0;
      overflow: auto;
      font-family: system-ui, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto,
      Oxygen, Ubuntu, Cantarell, "Open Sans", "Helvetica Neue", sans-serif;
     }

`layout.js`

    import './globals.css'
    import { Inter } from 'next/font/google'

    const inter = Inter({ subsets: ['latin'] })

    export const metadata = {
      title: 'Create Next App',
      description: 'Generated by create next app',
    }

    export default function RootLayout({ children }) {
      return (
        <html lang="en">
         <body className={inter.className}>{children}</body>
        </html>
     )
    }


after updating those files please restart the project and clear the browser cache.

**Comment by italobarrosme** [comment]
if weight is assigned, remove

example
```html 
const nunito = Nunito_Sans({
  weight: '400', ✖️
  subsets: ['latin'],
  display: 'swap',
  adjustFontFallback: false,
})`

**Reply by nalladev** [reply]
weight is a required param

**Comment by franva** [comment]
In the Nextjs tutorial, this error still happens and I'm not using a proxy.
The code I'm using is provided by the tutorial 
https://nextjs.org/learn/dashboard-app

Next.js 14.0.0
react: 18.2.0

**Comment by PiusLucky** [comment]
Late to the Party.

For others that might be having similar issue.

Remove your `.next `folder

`rm -rf .next`

Run build again

`yarn build`

Restart your server.

**Reply by absolutegravitas** [reply]
i have to do this every time... isnt there a better way...

**Reply by yusuphwickama** [reply]
Adding this as a RUN command on my Dockerfile worked!

**Comment by dalmasonto** [comment]
For my case it was the internet I was using. Somehow it was making next not to be able to download the font. After switching the internet, it worked fine.

**Comment by Cauen** [comment]
For me i've fixed removing the font declaration from a deep layout file
```
// src/app/dashboard/certificates/models/layout.tsx ❌
const fontRobotoSans = Roboto({
  weight: ['100', '300', '400', '500'],
  style: ['normal', 'italic'],
  subsets: ['latin'],
  variable: '--font-roboto',
  display: 'swap',
})
```

And export it from a src/app/fonts.ts file

``` 
/// src/app/fonts.ts ✅
import { Inter, Roboto } from 'next/font/google'

export const fontInterSans = Inter({
  subsets: ['latin'],
  variable: '--font-inter',
})

export const fontRobotoSans = Roboto({
  weight: ['100', '300', '400', '500'],
  style: ['normal', 'italic'],
  subsets: ['latin'],
  variable: '--font-roboto',
  display: 'swap',
})

```

**Comment by jmotes** [comment]
I'm having this issue when using the app router (Next.js 14.2.4) with Turbopack inside a docker container. I've tried `rm -rf .next` suggested but it doesn't help with `dev` mode. The fonts load just fine with `npm run start` though.

My error message:

```
Failed to download `Inter` from Google Fonts. Using fallback font instead.
 ✓ Compiled / in 28.2s
 ⚠ [next]/internal/font/google/inter_c4e516d6.module.css
Error while requesting resource
There was an issue establishing a connection while requesting https://fonts.googleapis.com/css2?family=Inter:wght@100..900&display=swap.
```

I've verified the issue goes away if I run the project outside of docker. Any ideas on how to solve the issue specifically for Turbopack+Docker?

**Reply by amdlemos** [reply]
I have exactly the same problem, outside of docker it works perfectly.

**Reply by jmotes** [reply]
If it helps you, I ended up downloading the variable font from Google and using the local font option:

https://nextjs.org/docs/pages/building-your-application/optimizing/fonts#local-fonts

This way works fine with Docker + Turbopack...

**Reply by jmotes** [reply]
Ohhh, I almost forgot! I opened a new issue for this: https://github.com/vercel/next.js/issues/67324

While setting up the minimal repro for the issue, I discovered that the Node alpine docker image works fine if switching to it is an option.

**Reply by amdlemos** [reply]
I managed to solve the problem, in my docker compose I added dns as below, it worked perfectly:

```
services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    dns:
      - 8.8.8.8
      - 8.8.4.4
```

**Reply by jmotes** [reply]
Nice! I wonder why this is only required when using Turbopack. Maybe it has it's own DNS resolver that this overrides somehow?

**Reply by jmotes** [reply]
@amdlemos would you mind sharing your findings on the [ticket](https://github.com/vercel/next.js/issues/67324) I created? Maybe it would help the team narrow down the issue.

**Reply by amdlemos** [reply]
I tested your repository from issue #67324 and here it was normal without DNS. After taking a closer look at this issue here I noticed that I am not using the turbopack. But from what I understand it was a timeout problem. The build of my application that took almost 7s dropped to a little over 2.5s. So I don't know how to explain exactly what was causing the problem. I believe it was the response time.

Look at the difference in build time with and without dns
![20240709_16h42m33s_grim](https://github.com/vercel/next.js/assets/2641369/6bc64105-5e7b-4235-84d9-45706ddba827)

**Comment by usrrname** [comment]
Same issue, not on VPN, have a named export of Inter.

**Comment by SadmanYasar** [comment]
I was facing this issue and restarted my laptop then removed .next and ran npm install then npm run build and it got fixed.
Macbook Pro (MacOS Sequoia 15.3.1)
Nextjs v14.2.0
Nodejs v20

**Comment by apuatcfbd** [comment]
I'm having the same issue, Following is the output
```log
> env-cmd -f .env.development next build
▲ Next.js 15.2.4

   Creating an optimized production build ...


Retrying 1/3...


Retrying 2/3...


Retrying 3/3...
[AggregateError: ] { code: 'ETIMEDOUT' }
Failed to compile.

src/app/layout.tsx
`next/font` error:
Failed to fetch `Inter` from Google Fonts.


> Build failed because of webpack errors
 ELIFECYCLE  Command failed with exit code 1.
```

Then I wanted to check if the website is reachable
```shell
> curl https://fonts.googleapis.com/css2?family=Inter&display=swap
[1]+  Done                    curl https://fonts.googleapis.com/css2?family=Inter
[1] 179498
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 400;
  src: url(https://fonts.gstatic.com/s/inter/v18/UcCO3FwrK3iLTeHuS_nVMrMxCp50SjIw2boKoduKmMEVuLyfMZg.ttf) format('truetype');
}
```
It means the website is reachable. So what? may b the font itself is not available?
```shell
> wget https://fonts.gstatic.com/s/inter/v18/UcCO3FwrK3iLTeHuS_nVMrMxCp50SjIw2boKoduKmMEVuLyfMZg.ttf
[1]+  Done                    curl https://fonts.googleapis.com/css2?family=Inter
--2025-05-15 18:11:20--  https://fonts.gstatic.com/s/inter/v18/UcCO3FwrK3iLTeHuS_nVMrMxCp50SjIw2boKoduKmMEVuLyfMZg.ttf
Resolving fonts.gstatic.com (fonts.gstatic.com)... 142.250.199.163, 2404:6800:4002:81b::2003
Connecting to fonts.gstatic.com (fonts.gstatic.com)|142.250.199.163|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 324820 (317K) [font/ttf]
Saving to: ‘UcCO3FwrK3iLTeHuS_nVMrMxCp50SjIw2boKoduKmMEVuLyfMZg.ttf’

UcCO3FwrK3iLTeHuS_nVMrMxCp50SjIw2bo 100%[=================================================================>] 317.21K  1.22MB/s    in 0.3s    

2025-05-15 18:11:21 (1.22 MB/s) - ‘UcCO3FwrK3iLTeHuS_nVMrMxCp50SjIw2boKoduKmMEVuLyfMZg.ttf’ saved [324820/324820]
```
It appears that the font itself also available. Why Next.js failing to build? There's no proxy / VPN in the machine.

FYI:
In my local machine it builds perfectly the OS is Ubuntu 24.04
In another PC (where the problem is occurring) it is not working, that machine is on CentOS 9
both are using same network !

**Comment by sarthaksharma27** [comment]
Facing the same issue so i clear my browser cache and build again and it's worked fine.

**Reply by apuatcfbd** [reply]
clearing browser cache making build successful !! where's the connection?

**Comment by monolithed** [comment]
Here's what worked for me:

```
export NODE_OPTIONS="--network-family-autoselection-attempt-timeout=500"
```

**Comment by Nzanzu-Lwanzo** [comment]
Hey y'all ! Adding proload:false to the font configuration worked for me.
```
export const outfit = outfitFont({
      variable: "--font-outfit",
      subsets: ["latin"],
      weight: ["300", "400", "700"],
      display: "swap",
      preload: false
});
```
